src/controller/CHIPDeviceController.cpp

```diff
@@ -740,6 +742,32 @@ CHIP_ERROR DeviceCommissioner::EstablishPASEConnection(NodeId remoteDeviceId, Re
     return err;
 }
 
+void DeviceCommissioner::OnDiscoveredDeviceOverBleSuccess(void * appState, BLE_CONNECTION_OBJECT connObj)
+{
+    auto self   = static_cast<DeviceCommissioner *>(appState);
+    auto device = self->mDeviceInPASEEstablishment;
+    if (nullptr != device)
+    {
+        auto remoteId = device->GetDeviceId();
+
+        RendezvousParameters params;
+        params.SetPeerAddress(Transport::PeerAddress::BLE()).SetConnectionObject(connObj);
+
+        self->ReleaseCommissioneeDevice(device);
+        LogErrorOnFailure(self->EstablishPASEConnection(remoteId, params));
+    }
+}
+
+void DeviceCommissioner::OnDiscoveredDeviceOverBleError(void * appState, CHIP_ERROR err)
+{
+    auto self   = static_cast<DeviceCommissioner *>(appState);
+    auto device = self->mDeviceInPASEEstablishment;
+    if (nullptr != device)
+    {
+        self->ReleaseCommissioneeDevice(device);
```

Comment:
Is it worth checking that the device's session is in fact a BLE session?  I'm a little worried about having these un-cancelable async bits, but I guess a lot of commissioning is like that.....